src/active_mock.rs

```diff
@@ -46,10 +46,18 @@ impl ActiveMock {
     pub(crate) fn verify(&self) -> bool {
```

Comment:
This method should probably return a `VerificationReport` :thinking: 
`VerificationReport` can then have a `is_satisfied` method returning a boolean which you can use in the `filter` call.

Reply:
There is a performance penalty: In this way you are creating VerificationReports for all the ActiveMocks, which involves a clone:

https://github.com/MarcoIeni/wiremock-rs/blob/89d7d1fcb72805b83c567e690441bb10101a656d/src/verification.rs#L16

If you filter before, you only create VerificationReports for not verified ActiveMocks.

If you don't care about this I can do your suggested change.

Reply:
I'd say we don't care - let's optimise for readability here.

Reply:
(If you want, I'd say there should be a way to avoid the clone by using references in VerificationReport)

Reply:
> I'd say we don't care - let's optimise for readability here.

done in a126306

> (If you want, I'd say there should be a way to avoid the clone by using references in VerificationReport)

I tried but I had problems, I will do a separate branch where I show you

Reply:
> I tried but I had problems, I will do a separate branch where I show you

Here is it: https://github.com/MarcoIeni/wiremock-rs/tree/expect-ref

Reply:
Argh, true - you'd have to hold on the read permit on the lock, which is indeed undesirable. Let's clone our way out, as you did.